src/inference/rejection.js

```diff
@@ -33,28 +41,49 @@ module.exports = function(env) {
     this.oldCoroutine = env.coroutine;
     env.coroutine = this;
 
-    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
-      throw new Error('samples should be a positive integer.');
-    }
-
     if (this.incremental && this.maxScore > 0) {
       util.warn('Rejection: Reduce maxScore to zero for better performance.');
     }
   }
 
   Rejection.prototype.run = function() {
+    if (!_.isNumber(this.numSamples) || this.numSamples <= 0) {
+      return this.error('"samples" should be a positive integer.');
+    }
+    var elapseSec = (Date.now() - this.startTime) / 1000.0;
+    if (elapseSec > 2) {
```

Comment:
I think the logic here is we only care about what happens in the first 2 seconds. If `sampleRate` is good then we'll let rejection execute till it finishes, rather than check every time. Is it better to set a flag here so that `if (elapseSec > 2) {...}` will be skipped later on?